OpenRA.Test/PathfinderTests.cs

```diff
@@ -0,0 +1,241 @@
+
+			IPathSearch search;
+			Stopwatch stopwatch;
+			List<CPos> path1 = null;
+			List<CPos> path2 = null;
+			List<CPos> path3 = null;
+			List<CPos> path4 = null;
+			List<CPos> path5 = null;
+			List<CPos> path6 = null;
+			List<CPos> path7 = null;
+			List<CPos> path8 = null;
+			var pathfinder = new PathFinder(world);
+
+			// Act
+			stopwatch = new Stopwatch();
+			foreach (var a in Enumerable.Range(1, 50))
+			{
+				search = PathSearch.FromPoint(world, mi, actor, from, target, true);
+				stopwatch.Start();
+				path5 = pathfinder.FindPath(search);
+
+				stopwatch.Stop();
+				search = PathSearch.FromPoint(world, mi, actor, new CPos(0, 0), new CPos(51, 100), true);
+				stopwatch.Start();
+				path6 = pathfinder.FindPath(search);
+
+				stopwatch.Stop();
+				search = PathSearch.FromPoint(world, mi, actor, new CPos(0, 0), new CPos(49, 50), true);
+				stopwatch.Start();
+				path7 = pathfinder.FindPath(search);
+
+				stopwatch.Stop();
+				search = PathSearch.FromPoint(world, mi, actor, new CPos(127, 0), new CPos(50, 101), true);
+				stopwatch.Start();
+				path8 = pathfinder.FindPath(search);
+			}
+
+			Console.WriteLine("I took " + stopwatch.ElapsedMilliseconds + " ms with new pathfinder");
+
+			IPathSearch search2;
+			stopwatch = new Stopwatch();
+			foreach (var a in Enumerable.Range(1, 50))
+			{
+				search = PathSearch.FromPoint(world, mi, actor, from, target, true);
+				search2 = PathSearch.FromPoint(world, mi, actor, target, from, true).Reverse();
+				stopwatch.Start();
+				path5 = pathfinder.FindBidiPath(search, search2);
+
+				stopwatch.Stop();
+				search = PathSearch.FromPoint(world, mi, actor, new CPos(0, 0), new CPos(51, 100), true);
+				search2 = PathSearch.FromPoint(world, mi, actor, new CPos(51, 100), new CPos(0, 0), true).Reverse();
+				stopwatch.Start();
+				path6 = pathfinder.FindBidiPath(search, search2);
+
+				stopwatch.Stop();
+				search = PathSearch.FromPoint(world, mi, actor, new CPos(0, 0), new CPos(49, 50), true);
+				search2 = PathSearch.FromPoint(world, mi, actor, new CPos(49, 50), new CPos(0, 0), true).Reverse();
+				stopwatch.Start();
+				path7 = pathfinder.FindBidiPath(search, search2);
+
+				stopwatch.Stop();
+				search = PathSearch.FromPoint(world, mi, actor, new CPos(127, 0), new CPos(50, 101), true);
+				search2 = PathSearch.FromPoint(world, mi, actor, new CPos(50, 101), new CPos(127, 0), true).Reverse();
+				stopwatch.Start();
+				path8 = pathfinder.FindBidiPath(search, search2);
+			}
+
+			Console.WriteLine("I took " + stopwatch.ElapsedMilliseconds + " ms with new FindBidipathfinder");
+		}
+
+		private static int Est1(CPos here, CPos destination)
+		{
+			var diag = Math.Min(Math.Abs(here.X - destination.X), Math.Abs(here.Y - destination.Y));
+				var straight = Math.Abs(here.X - destination.X) + Math.Abs(here.Y - destination.Y);
+
+				// Min cost to arrive from once cell to an adjacent one
+				// (125 according to tests)
+				const int D = 100;
```

Comment:
That's some Resharper (a visual studio refactor tool) recommendations. It's no big deal.